test/node/uuid_tests.js

```diff
@@ -162,4 +162,42 @@ describe('UUID', () => {
     const uuid = new UUID(UPPERCASE_DASH_SEPARATED_UUID_STRING);
     expect(inspect(uuid)).to.equal(`new UUID("${LOWERCASE_DASH_SEPARATED_UUID_STRING}")`);
   });
+
+  describe('deserialize', () => {
+    const serializedUUID = BSON.serialize({ uuid: new BSON.UUID().toBinary() });
+    it('should promoteUUIDs when flag is true', () => {
+      const { uuid: promotedUUID } = BSON.deserialize(serializedUUID, { promoteUUIDs: true });
+      expect(promotedUUID._bsontype).to.equal('UUID');
+    });
+
+    it('should not promoteUUIDs when flag is false', () => {
+      const { uuid: unpromotedUUID } = BSON.deserialize(serializedUUID, { promoteUUIDs: false });
+      expect(unpromotedUUID._bsontype).to.equal('Binary');
+    });
+
+    it('should not promoteUUIDs when flag is omitted', () => {
+      const { uuid: omittedFlagUUID } = BSON.deserialize(serializedUUID);
+      expect(omittedFlagUUID._bsontype).to.equal('Binary');
+    });
```

Comment:
Discussed adding a test to assert that the toUUID errors will also get thrown from the deserializer when those unhappy paths are encountered

Reply:
Sounds good!